File: java/MsgClient.java
// CS 6421 - Simple Message Board Client in Java
// T. Wood
// Compile with: javac MsgClient
// Run with:     java MsgClient

// @author Cong Sun
// @Gwid G48099998

import java.io.IOException;
import java.io.PrintWriter;
import java.net.Socket;
import java.net.UnknownHostException;

public class MsgClient {
    public static void main(String[] args) {
        //System Argument should be [Host IP] [Port Number] [Message_Name] [Message_Content], check the count for the argument before try to run the program
        if(args.length != 4){
            System.out.println("argument number not corrent");
            System.exit(-1);
        }
        
        //The First Argument is the Host IP
        String host = args[0];
        //The Second Argument is the Port Number
        int portnum = Integer.parseInt(args[1]);
        try{
            //Create the socket object
            Socket socket = new Socket(host,portnum);
            //Create the PrintWriter object using the socket object
            PrintWriter out = new PrintWriter(socket.getOutputStream(), true);
            
            //Send [Message_Name] and a new line character
            out.println(args[2]);
            //Send [Message_Content]
            out.println(args[3]);
        }catch(Exception e){
            System.out.println(e);
        }

    }
}

File: python/msgclient.py
# CS 6421 - Simple Message Board Client in Python
# T. Wood
# Run with:     python msgclient.py

# @author Cong Sun
# @Gwid G48099998

import socket
import sys


# System Argument should be [fileName] [Host IP] [Port Number] [Message_Name] [Message_Content], check the count for the argument before try to run the program
if len(sys.argv) != 5: print("error")

# The Second Argument is the Host IP
host = sys.argv[1];
#The Third Argument is the Port Number
portnum = sys.argv[2];

#Create the socket object
clientsocket = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
#Connect to the server using the host IP and Port number
clientsocket.connect((host,int(portnum)))

#Send [Message_Name]
clientsocket.send(sys.argv[3])
#Send a new line character
clientsocket.send("\n")
#Send [Message_Content]
clientsocket.send(sys.argv[4])
print("Sent message to server!")
